test/test_utilities.cpp

```diff
@@ -85,6 +85,40 @@ void AssignInputValuesRandom(Tensor& A) {
   }
 }
 
+template <>
+void AssignInputValuesRandom(Tensor& A, int32_t min, int32_t max) {
+  auto A_flat = A.flat<int>();
+  auto A_flat_data = A_flat.data();
+  srand(static_cast<unsigned>(time(0)));
+  for (int i = 0; i < A_flat.size(); i++) {
+    // give a number between 0 and max
+    int32_t value = static_cast<int32_t>(rand()) /
+                    static_cast<int32_t>(RAND_MAX / (int32_t)(max - min));
+    value = value + min;  // range from min to max
+    // value =
+    //     roundf(value * 100) / 100.0;  // change the precision of the float to
+    //                                   // 2 number after the decimal
+    A_flat_data[i] = value;
+  }
+}
+
+template <>
+void AssignInputValuesRandom(Tensor& A, int64_t min, int64_t max) {
```

Comment:
Trying to understand the use-cases for specialized template.

1. We want to round the precision to two decimals for float data types? In which case it might be better to add the specialized template for float. As it is doing something more than the regular template.

2.  Do we expect explicit function calls for int32_t and int64_t?
And map both int32 and int64 to int , here ? 
``` auto A_flat = A.flat<int>();```

3. For explicitly assigning, int32 for DT_INT32 and int64 for DT_INT64? Which might not happen the way we have defined the template. I think

Reply:
Thanks for the detailed comments. Actually this is the old version of my change, I am sorry it lands here. I will push the latest change. 

1. You are right, we can create a special template for float and put those two lines there. 

2. I did separate int32 and int64 because for some inputs, in the documentation, it might say it can have type int32 or int64, so in that case, we can test different situation for different int types.